src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
                 },
                 _joinableFactory);
 
+            _solutionLoadedEvent = new AsyncManualResetEvent(initialState: true);
```

Comment:
Yes, subsequent call to `Reset` will make things right by resetting the event. So it's not important how it's constructed. The only reason I passed _true_ here was this remark on [msdn](https://msdn.microsoft.com/en-us/library/microsoft.visualstudio.threading.asyncmanualresetevent.asyncmanualresetevent.aspx):

> **false** better simulates the behavior of theManualResetEventSlim class, but **true** can result in slightly better performance.